src/components/Expenses/ExpensesAnalysisChart.jsx

```diff
@@ -0,0 +1,174 @@
+  function setOptionSelect(responseArray) {
+    responseArray.forEach((response) => {
+      yearArray.push(response.date.split("-")[0]);
+    });
+    setYearOfExpense([...new Set(yearArray)]);
+  }
+
+  // setGraphData is responsible for converting each expense into
+  // viable data structure/format to be used in react-chartjs-2
+  // Each expense is created in an array of length 12, mostly filled with 0
+  // and within one of the index, is where the expense data is positioned.
+  // Positioning is based on which month the expense is created, from 0-11
+  // which represents january - dec.
+  function setGraphData(responseData, year) {
+    const tempArray = [];
+    const tempData = [];
+    responseData.map((response) => {
+      if (response.date.split("-")[0] === year) {
+        // This is to push data that corresponds to the year selected. Later used for populating title.
+        tempData.push(response); 
+        switch (response.date.split("-")[1]) {
+          case "01":
+            tempArray.push([response.amount, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]);
+            break;
+          case "02":
+            tempArray.push([0, response.amount, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]);
+            break;
+          case "03":
+            tempArray.push([0, 0, response.amount, 0, 0, 0, 0, 0, 0, 0, 0, 0]);
+            break;
+          case "04":
+            tempArray.push([0, 0, 0, response.amount, 0, 0, 0, 0, 0, 0, 0, 0]);
+            break;
+          case "05":
+            tempArray.push([0, 0, 0, 0, response.amount, 0, 0, 0, 0, 0, 0, 0]);
+            break;
+          case "06":
+            tempArray.push([0, 0, 0, 0, 0, response.amount, 0, 0, 0, 0, 0, 0]);
+            break;
+          case "07":
+            tempArray.push([0, 0, 0, 0, 0, 0, response.amount, 0, 0, 0, 0, 0]);
+            break;
+          case "08":
+            tempArray.push([0, 0, 0, 0, 0, 0, 0, response.amount, 0, 0, 0, 0]);
+            break;
+          case "09":
+            tempArray.push([0, 0, 0, 0, 0, 0, 0, 0, response.amount, 0, 0, 0]);
+            break;
+          case "10":
+            tempArray.push([0, 0, 0, 0, 0, 0, 0, 0, 0, response.amount, 0, 0]);
+            break;
+          case "11":
+            tempArray.push([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, response.amount, 0]);
+            break;
+          case "12":
+            tempArray.push([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, response.amount]);
+            break;
+        }
+      }
+    });
+    formatGraphData(tempArray, tempData); // Formatted graph array and data passed
+  }
+
+  // formatGraphData is used to format and return the correct format and extra 
+  // options that are needed to pass to react-chartjs-2, such as background colour 
+  // and labels. Count is used to correspond with the array and data to populate each 
+  // labels
+  function formatGraphData(array, data) {
+    let count = 0;
+    const final = [];
+    array.forEach((item) => {
```

Comment:
You could make this into a .map with index for the count, would save the count variable. Would just be a little cleaner, no biggy